src/lib/y2network/widgets/additional_addresses.rb

```diff
@@ -154,124 +155,16 @@ def handle(event)
-              TextEntry(Id(:netmask), _("Net&mask"), mask)
-            ),
-            HSpacing(1),
-            HBox(
-              PushButton(Id(:ok), Opt(:default), Yast::Label.OKButton),
-              PushButton(Id(:cancel), Yast::Label.CancelButton)
-            )
-          )
-        )
-
-        Yast::UI.ChangeWidget(
-          Id(:name),
-          :ValidChars,
-          Yast::String.CAlnum
-        )
-        Yast::UI.ChangeWidget(Id(:ipaddr), :ValidChars, Yast::IP.ValidChars)
-
-        if entry
-          Yast::UI.SetFocus(Id(:ipaddr))
-        else
-          Yast::UI.SetFocus(Id(:name))
-        end
-
-        while (ret = Yast::UI.UserInput) == :ok
-
-          res = {}
-          val = Yast::UI.QueryWidget(Id(:name), :Value)
-
-          if "#{name}.#{val}" !~ /^[[:alnum:]._:-]{1,15}\z/
-            # Popup::Error text
-            Yast::Popup.Error(_("Label is too long."))
-            Yast::UI.SetFocus(Id(:name))
-            next
-          end
+    private
 
-          res[:label] = val
-
-          ip = Yast::UI.QueryWidget(Id(:ipaddr), :Value)
-          if !Yast::IP.Check(ip)
-            # Popup::Error text
-            Yast::Popup.Error(_("The IP address is invalid."))
-            Yast::UI.SetFocus(Id(:ipaddr))
-            next
-          end
-          res[:ip] = ip
-
-          val = Yast::UI.QueryWidget(Id(:netmask), :Value)
-          if !valid_prefix_or_netmask(ip, val)
-            # Popup::Error text
-            Yast::Popup.Error(_("The subnet mask is invalid."))
-            Yast::UI.SetFocus(Id(:netmask))
-            next
-          end
-
-          netmask = ""
-          prefixlen = ""
-          if val.start_with?("/")
-            prefixlen = val[1..-1]
-          elsif Yast::Netmask.Check6(val)
-            prefixlen = val
-          else
-            netmask = val
-          end
-
-          res[:prefixlen] = if prefixlen.empty?
-            IPAddr.new("#{netmask}/#{netmask}")
-          else
-            prefixlen
-          end
-          res[:prefixlen] = prefixlen
-          res[:id] = id
-
-          break
-        end
-
-        Yast::UI.CloseDialog
-        return nil if ret != :ok
-
-        res
-      end
-
-      def valid_prefix_or_netmask(ip, mask)
-        valid_mask = false
-        mask = mask[1..-1] if mask.start_with?("/")
-
-        if Yast::IP.Check4(ip) && (Yast::Netmask.Check4(mask) || Yast::Netmask.CheckPrefix4(mask))
-          valid_mask = true
-        elsif Yast::IP.Check6(ip) && Yast::Netmask.Check6(mask)
-          valid_mask = true
-        else
-          log.warn "IP address #{ip} and mask #{mask} is not valid"
-        end
-        valid_mask
+      # Convenience method to obtain an object with the additionl IP address
+      # data.
+      #
+      # @param index [Integer, nil] the position of the alias to be edited or
+      #   nil in case a new one is wanted
+      # @return [OpenStruct] additional IP address data
+      def settings_struct_for(index)
```

Comment:
I think that method name can be without struct so just `settings_for` or even `settings`